### ISSUE TITLE
Bug: Unable to import files in Linux

### ISSUE THREAD

**Issue by JerJohn15** [octaviospain/Musicott#6]
So I've recently got this application to work by forking and building it via GitHub. However, when I attempt to import an audio file either by file or folders, I'm unable to see any files (currently, mp3, flac, and wav), like @octaviospain  has reported in #3. My "Musicott-tracks.json"  and "Musicott-main-log.txt
 files return very similar contents when I open them:

Musicott-tracks.json:

``` json

  {
  "@type":"com.sun.javafx.collections.ObservableMapWrapper"
}
```

``` json

java.util.prefs run 29/10/16 08:26:09 :831000000
0   INFO:Created user preferences directory.

com.transgressoft.musicott.MusicottApplication loadConfigProperties 29/10/16 08:26:24 :327000000
1   WARNING:Error loading configuration properties

java.io.FileNotFoundException: resources/config/config.properties (No such file or directory)
java.io.FileInputStream.open0(Native Method)
java.io.FileInputStream.open(FileInputStream.java:195)
java.io.FileInputStream.<init>(FileInputStream.java:138)
java.io.FileInputStream.<init>(FileInputStream.java:93)
com.transgressoft.musicott.MusicottApplication.loadConfigProperties(MusicottApplication.java:177)
com.transgressoft.musicott.MusicottApplication.init(MusicottApplication.java:154)
com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:841)
com.sun.javafx.application.LauncherImpl.lambda$launchApplication$155(LauncherImpl.java:182)
java.lang.Thread.run(Thread.java:745)
javafx.css calculateValue 29/10/16 08:26:25 :782000000
2   WARNING:Caught 'java.lang.ClassCastException: java.lang.String cannot be cast to javafx.scene.paint.Color' while converting value for '-fx-background-color' from rule '*.progress-bar>*.track' in stylesheet jar:file:/usr/lib/jvm/java-8-oracle/jre/lib/ext/jfxrt.jar!/com/sun/javafx/scene/control/skin/modena/modena.bss

```

I started importing just a music album with .mp3 files in them which caused this bug. Then I checked to make sure that all of the file types are [selected](https://github.com/octaviospain/Musicott/wiki/Importing-files) at  "File->Preferences". After trying to re-import everything again, I was still unable to see anything in the player. 

**Update:**

As another test, I've tried to import the test files located in `Musicott/test-resources/testfiles` and I still get the same problem.

**Comment by octaviospain** [comment]
I can't reproduce this bug. Which Java version are you using? Under which operative system?

I just pushed a hot fix https://github.com/octaviospain/Musicott/commit/2189b41c39490ca721fdc5e90f77807b453f04b4 to the develop branch including log messages through the user case and a slight modification. 
Pull it, delete the `Musicott`folder form your home folder if there is already one, modify `Musicott/resources/config/logging.properties`so it contains:
```properties
org.jaudiotagger.tag.id3.level=OFF
org.jaudiotagger.tag.mp4.level=OFF
org.jaudiotagger.audio.level=OFF
be.tarsos.transcoder.level=OFF
com.transgressoft.musicott.level=OFF
com.transgressoft.musicott.tasks.level=ALL
com.transgressoft.musicott.util.level=ALL
com.transgressoft.musicott.view.custom.level=ALL
```
and perform the following steps:

1. Run `Musicott` from the `develop` branch
2. Click on `File > Preferences` and check that `mp3, m4a, wav` and `flac` are selected in `Import file formats`
3. Click on the `File > Import Folder...` menu
4. Navigate and select `Musicott/test-resources/testfiles` folder
5. Accept the confirmation alert

Paste here the content of the `Musicott-main-log.txt`file.

**Comment by JerJohn15** [comment]
@octaviospain ,

I'm running Java 1.8.0_111 on a Linux Mint 17.3 virtual machine.  

I've tried performing the steps you've outlined but I'm still running into the same problem. 
My ```Musicott/resources/config/logging.properties``` is identical to the one you have above and I've made sure I deleted my Musicott folder in addition to being on the ```dev branch```. Here is my log output file:

```json
com.transgressoft.musicott.tasks.SaveMusicLibraryTask serializeTracks 05/11/16 15:43:47 :743000000
0	FINE:Saved list of tracks in /home/jcoffee/Music/Musicott/Musicott-tracks.json

com.transgressoft.musicott.view.custom.MusicottMenuBar defaultMenuBar 05/11/16 15:43:48 :196000000
1	FINE:Default menu bar created

javafx.css calculateValue 05/11/16 15:43:48 :398000000
2	WARNING:Caught 'java.lang.ClassCastException: java.lang.String cannot be cast to javafx.scene.paint.Color' while converting value for '-fx-background-color' from rule '*.progress-bar>*.track' in stylesheet jar:file:/usr/lib/jvm/java-8-oracle/jre/lib/ext/jfxrt.jar!/com/sun/javafx/scene/control/skin/modena/modena.bss

com.transgressoft.musicott.view.custom.MusicottMenuBar lambda$setFileMenuActions$2 05/11/16 15:45:20 :621000000
3	FINE:Choosing folder to being imported

com.transgressoft.musicott.tasks.ParseTask call 05/11/16 15:45:27 :302000000
4	FINE:Starting file importing

```

I should mention that I've made my default Musicott folder to be inside my Music folder and not at the original default location the application puts it in (I believe that was Home/[your username]/Musicott).

**Comment by octaviospain** [comment]
Hello again @JerJohn15, sorry for the late answer. I was trying to figure why this problem occurs, but I got no chance, and as I'm going to explain, this bug almost breaks my head...

I was able to reproduce this bug using a Linux Mint 17.3 virtual machine using Java 1.8.0_111, but I almost didn't find any _clues_ of this very strange bug:

Debugging the application **I found that the `ParseTask` thread object that has to perform the file import, stops it execution at [line 96](https://github.com/octaviospain/Musicott/blob/master/src/com/transgressoft/musicott/tasks/ParseTask.java#L96)**, when `MetadataParser.createTrack(file)` is called in order to parse a `File` into a `Track` object. **With no apparent reason.**

```java
private Track parseFileToTrack(File file) {
	Track newTrack = null;
	try {
		newTrack = MetadataParser.createTrack(file);      // line 96
		LOG.debug("Parsed file {}: {}", file, newTrack);
	}
	catch (TrackParseException exception) {
		LOG.error("Error parsing file {}: ", file, exception);
		parseErrors.add(file + ": " + exception.getMessage());
	}
	return newTrack;
}
```

Firstly I figured that the problem might be with the `MetadataParser` class, but the test [MetadataParserTest](https://github.com/octaviospain/Musicott/blob/master/test/com/transgressoft/musicott/util/MetadataParserTest.java "MetadataParserTest") runs without problems...

So I began to try some modifications of the code:

* I created a constructor in `MetadataParser` with no parameters and a method with random code inside.
* I created a copy of `MetadataParser` doing the same thing but without a `static` `createTrack` function. I created a constructor passing the `File` and a method that create the `Track`.
* I placed the call to `MetadataParser.createTrack(file)` in different lines
* I placed random code of creation & usage of random objects before the calling of `MetadataParser.createTrack(file)`

**With no different result. The execution of the `ParseTask` thread finishes when it lands to the `MetadataParser.createTrack(file)` statement.**

I also reproduced this bug with the same behaviour in an Ubuntu 16.04.1 virtual machine, so it looks like this _black magic_ bug occurs in Linux, because this don't happend using Musicott on Windows or OS X.
